src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -255,48 +261,56 @@ export const txHistoryApi = createApi({
         const results: TransactionsByAccountId = {}
         await Promise.all(
           accountIds.map(async accountId => {
+            results[accountId] = []
+
             const { chainId, account: pubkey } = fromAccountId(accountId)
             const adapter = getChainAdapterManager().get(chainId)
-            if (!adapter)
-              return {
-                error: {
-                  data: `getAllTxHistory: no adapter available for chainId ${chainId}`,
-                  status: 400,
-                },
-              }
 
-            let currentCursor: string = ''
+            if (!adapter) {
+              const data = `getAllTxHistory: no adapter available for chainId ${chainId}`
+              return { error: { data, status: 400 } }
+            }
 
             try {
+              let currentCursor = ''
               do {
+                const pageSize = (() => {
+                  switch (chainId) {
+                    case gnosisChainId:
+                    case polygonChainId:
+                      return 10
```

Comment:
Reduced page size should help with failed request rate on chains with high amount of spam token/nft transactions